js_modules/dagster-ui/packages/ui-core/src/runs/AssetTagCollections.tsx

```diff
@@ -180,7 +194,7 @@ export const AssetKeyTagCollection = React.memo((props: AssetKeyTagCollectionPro
         overflow: 'hidden',
       }}
     >
-      {assetKeys.slice(0, rendered).map((assetKey) => (
+      {sortedAssetKeys.slice(0, rendered).map((assetKey) => (
```

Comment:
this `slice(0, rendered)` is going to create a new array on every render. Better to just include this slice as part of the memo on line 161.